src/pyfreeleh/row/gsheet.py

```diff
@@ -1,412 +1,458 @@
+        """Filter the rows that we're going to get.
+
+        The given `condition` will be used as the WHERE clause on the final query. You can use `"?"` placeholder
+        inside the condition and will be replaced with the actual value given in the `*args` variadic parameter
+        based on their appearance ordering.
+
+        Args:
+            condition: conditions of the data that we're going to get.
+            *args: list of arguments that will be used to fill in the placeholders in the given `condition`.
+
+        Returns:
+            SelectStmt: the select statement with the given WHERE condition applied.
+
+        Examples:
+            To apply "WHERE age > 10" filter on the select statement:
+
+            >> len(store.select().where("age > ?", 10).execute())
+            10
+        """
+        mapped_conditions = map_conditions(self._mapper, condition)
+        self._query.where(mapped_conditions, *args)
         return self
 
-    def limit(self, limit: int) -> "SelectStmt":
+    def limit(self, limit: int) -> "SelectStmt[T]":
+        """Defines the maximum number of rows that we're going to return.
+
+        Args:
+            limit: limit that we want to apply.
+
+        Returns:
+            SelectStatement: select statement with the limit applied.
+        """
         self._query.limit(limit)
         return self
 
-    def offset(self, offset: int) -> "SelectStmt":
+    def offset(self, offset: int) -> "SelectStmt[T]":
+        """Defines the offset of the returned rows.
+
+        Args:
+            offset: offset that we want to apply.
+
+        Returns:
+            SelectStatement: select statement with the offset applied.
+        """
         self._query.offset(offset)
         return self
 
-    def order_by(self, **kwargs: Ordering) -> "SelectStmt":
-        self._query.order_by(**kwargs)
+    def order_by(self, *orderings: Ordering) -> "SelectStmt[T]":
+        """Defines the column ordering of the returned rows.
+
+        Args:
+            *orderings: the column ordering that we want to apply.
+
+        Returns:
+            SelectStatement: select statement with the column ordering applied.
+        """
+        mapped_orderings = map_orderings(self._mapper, list(orderings))
+        self._query.order_by(*mapped_orderings)
         return self
 
-    def execute(self) -> List[Dict[str, Any]]:
-        rows = self._wrapper.query(
-            self._spreadsheet_id,
-            self._sheet_name,
-            self._query.build_select(self._selected_columns),
-        )
+    def execute(self) -> List[T]:
+        """Execute the select statement.
+
+        Returns:
+            list: list of rows that matched the given condition.
+        """
+        mapped_columns = []
+        for col in self._selected_columns:
+            mapped_columns.append(self._mapper.to_column(col))
+
+        if "A" not in mapped_columns:
+            mapped_columns.insert(0, "A")
```

Comment:
Why do you need the `_rid` column here?

Reply:
So that I can set the value of `_rid` to the returned rows.

Client can use this info to tell whether the current instance is inserted already or not. Client can also do various stuff with this value, e.g. creating relations between rows, etc (but they need to be aware that the rid might be reused -> because of overwrite append).

Reply:
I think you are making a very strong assumption that we never discussed here. We never discussed whether the concept of a primary key should exist in the first place. I don't think we should expose this because it's hard to understand from the clienty perspective.

> but they need to be aware that the rid might be reused -> because of overwrite append

This is one strong reason why the client should not know about it. Another is the assumption about the location of the PK column.

If they want to have an ID system (e.g. to create the relations), then they should create their own ID system for now. Our PK "looks" like a PK, but it's not a PK because it's repeatable as you said.

Reply:
**Agreement:**

1. `_rid` is just a metadata. We should never treat this as an externally available data for clients to use directly.
2. We agreed that clients may (most likely will) need an ID system eventually.
     - This is the reason why you suggested using `_rid` as the ID (but the data the ID refers to can change due to deletion, which is not ideal for the client to always understand because we are trying to mimic SQL PK system).
     - However, as we agreed on using `_rid` as metadata only, we will need the client to provide their own system.
3. When the clients do have such requirements, then we agreed that we can provide our own implementation of the PK ID system that is available for the clients to use "somehow" (to be discussed how to implement it later). This can be using UUIDv4 or Snowflake internally (we can let the clients to choose).

Reply:
Done.